Realm/RLMObjectSchema.mm

```diff
@@ -285,11 +285,8 @@ - (BOOL)isEqualToObjectSchema:(RLMObjectSchema *)objectSchema {
     NSArray *otherProperties = objectSchema.properties;
     for (NSUInteger i = 0; i < _properties.count; i++) {
         RLMProperty *p1 = _properties[i], *p2 = otherProperties[i];
-        if (p1.type != p2.type ||
```

Comment:
But we do want to check for `isPrimary`? Can you explain why, I don't understand the distinction wrt accessors.